pydicom/pixel_data_handlers/gdcm_handler.py

```diff
@@ -161,20 +162,34 @@ def create_image(dicom_dataset, data_element):
     return image
 
 
-def create_image_reader(filename):
+def create_image_reader(ds):
     """Return a ``gdcm.ImageReader``.
 
     Parameters
     ----------
-    filename: str or unicode
-        The path to the DICOM dataset.
+    ds : pydicom.dataset.Dataset
+        The dataset to create the reader from.
 
     Returns
     -------
     gdcm.ImageReader
     """
     image_reader = gdcm.ImageReader()
-    image_reader.SetFileName(filename)
+    fname = getattr(ds, 'filename', None)
+    if fname and isinstance(fname, str):
+        pass
+    else:
+        # Copy the relevant elements and write to a temporary file to avoid
+        #   having to deal with all the possible objects the dataset may
+        #   originate with
+        new = ds.group_dataset(0x0028)
+        new["PixelData"] = ds["PixelData"]  # avoid ambiguous VR
```

Comment:
I don't understand that comment

Reply:
If I'd done `new.PixelData = ds.PixelData` then the VR would be 'OB or OW'

Reply:
I suppose its not so much avoiding it, as not making it worse

Reply:
Ah, ok, thanks!